core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -49,8 +50,13 @@ void imx_configure_tzasc(void)
 			     TZC_ATTR_SP_S_RW, region);
 		region = tzc_auto_configure(CFG_SHMEM_START, CFG_SHMEM_SIZE,
 			     TZC_ATTR_SP_ALL, region);
-		if (tzc_regions_lockdown() < 0)
+		ret = tzc_regions_lockdown();
+		if (ret != TEE_ERROR_SECURITY) {
 			EMSG("Region lockdown failed!");
+			panic();
```

Comment:
You could delete the `EMSG()` and do `panic("Region lockdown failed!");` instead

Reply:
for info: `panic("Region lockdown failed!");` prints the debug trace only upon `CFG_TEE_CORE_DEBUG=y` whereas `EMSG()` doesn't care about `CFG_TEE_CORE_DEBUG` config. Depends on which trace you expect...

Reply:
I'd like to keep the `EMSG`, I don't like systems which are completely silent even if the boot can't continue due to a `panic`.

Reply:
@etienne-lms you're right about `CFG_TEE_CORE_DEBUG` and I think it was a bad choice. There is a way to prevent any message on the secure console already: set `CFG_TEE_CORE_LOG_LEVEL` to 0. I agree with @Emantor, I think that as long as log level is not zero, we should always print something on panic. In non-debug mode we should perhaps just remove the file/line information, but keep the message.

I can propose a patch.

Reply:
Actually `panic()` always ends up calling `EMSG_RAW()` ([here])(https://github.com/OP-TEE/optee_os/blob/3.8.0/core/kernel/panic.c#L22-L28) so what I said above is wrong. I think we should just do this to not lose the message:
```diff
diff --git a/core/include/kernel/panic.h b/core/include/kernel/panic.h
index 523fdae5..b9a56dc2 100644
--- a/core/include/kernel/panic.h
+++ b/core/include/kernel/panic.h
@@ -12,7 +12,7 @@
 #if defined(CFG_TEE_CORE_DEBUG)
 #define __panic(str)   __do_panic(__FILE__, __LINE__, __func__, str)
 #else
-#define __panic(str)   __do_panic((void *)0, 0, (void *)0, (void *)0)
+#define __panic(str)   __do_panic((void *)0, 0, (void *)0, str)
 #endif
 
 void __do_panic(const char *file, const int line, const char *func,
```
...then `panic("Region lockdown failed!");` with no additional `EMSG()` would be fine.

Opinions?

Reply:
sounds good, I can add this as a patch.

Reply:
https://github.com/OP-TEE/optee_os/pull/3612

Reply:
good idea. thanks @jforissier.